numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -388,7 +395,15 @@ def _prepare_args(self, ty, val, stream, retr, kernelargs):
 
             c_intp = ctypes.c_ssize_t
 
-            meminfo = ctypes.c_void_p(0)
+            # hack: allocate too much space to hold the meminfo and set it all to 1
+            dbuf = cuda.to_device(np.ones(1024, dtype=np.uint8))
+            
+            # hack: don't garbage collect up the array containing the meminfo
+            def inner():
+                print(dbuf)
+            retr.append(inner)
+
+            meminfo = ctypes.c_void_p(dbuf.gpu_data._mem.device_ctypes_pointer.value)
```

Comment:
I am thinking this whole block of changes should really be a call to `NRT_MemInfo_new` building around the pointer to the array data

Reply:
Would you be calling it on the host and copying the initialized struct to the GPU?